casetokens.php

```diff
@@ -186,31 +228,49 @@ function casetokens_civicrm_tokenvalues(&$values, $cids, $job = NULL, $tokens =
     ));
     $clients = implode(', ', CRM_Utils_Array::collect('contact_id.display_name', $caseContact['values']));
 
-    // Get contacts from case roles
-    $relations = civicrm_api3('Relationship', 'get', array(
-      'case_id' => $caseId,
-      'options' => array('limit' => 0),
-      'is_active' => 1,
-      'contact_id_a.is_deleted' => 0,
-      'return' => array('relationship_type_id.name_b_a', 'contact_id_b'),
-    ));
+    $tomorrow = date('Y-m-d', strtotime(' +1 day'));
+    $today = date('Y-m-d', time());
+
+    $query = "SELECT crt.name_b_a, cr.contact_id_b " .
+      "FROM civicrm_relationship cr " .
+      "INNER JOIN civicrm_relationship_type crt ON cr.relationship_type_id = crt.id " .
+      "INNER JOIN civicrm_contact cc ON cr.contact_id_b = cc.id " .
+      "WHERE cr.is_active = 1 AND cr.case_id = $caseId AND cc.is_deleted = 0 " .
+      "AND ((cr.start_date < '$tomorrow' OR cr.start_date IS NULL) AND (cr.end_date > '$today' OR cr.end_date IS NULL)) " .
```

Comment:
Is there any reason why we are switching from the API to use an SQL query?

Also for fetching active relationships the correct AND condition here should be 
```php
AND ((cr.start_date <= '$today' OR cr.start_date IS NULL) AND (cr.end_date >= '$today' OR cr.end_date IS NULL)) "
```
No need to create an extra variable `$tomorrow` and also if the end date is today, the relationship is still valid.

You can see a sample query here: https://github.com/civicrm/org.civicrm.contactlayout/blob/master/CRM/Contactlayout/Helper/ProfileRelatedContact.php#L46-L57